staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -562,6 +563,54 @@ func (mounter *SafeFormatAndMount) formatAndMountSensitive(source string, target
 	return nil
 }
 
+func (mounter *SafeFormatAndMount) format(fstype string, args []string) ([]byte, error) {
+	mounter.acquireFormatSem()
+
+	out := make(chan *formatResult, 1)
+	go mounter.execFormat(fstype, args, out)
+	var res *formatResult
+
+	select {
+	case res = <-out:
+	case <-time.After(mounter.formatTimeout):
+		mounter.releaseFormatSem()
+	}
+
+	if res != nil {
+		// We received a result and still hold a semaphore slot.
+		mounter.releaseFormatSem()
+	} else {
+		// We timed out and released the semaphore slot.
```

Comment:
It took me a couple of readings to grok this, maybe the comment could read something like "we timed out and released the semaphore, but will continue to block until the format command finishes"

Reply:
Good idea. Done.